vividus-plugin-mobile-app/src/test/java/org/vividus/mobileapp/steps/TouchStepsTests.java

```diff
@@ -256,13 +261,43 @@ void shouldNotSwipeToElementIfItAlreadyExistsAndDoNotAdjustItForHorizontalScroll
         initSwipeMocks();
         WebElement element = mock(WebElement.class);
         mockAssertElementsNumber(List.of(element), true);
-
+        when(element.getRect()).thenReturn(new Rectangle(10, 500, 10, 10));
+        WebElement searchContext = mock(WebElement.class);
+        Point point = new Point(5, 5);
+        Dimension contextDimension = new Dimension(1920, 1080);
+        when(searchContext.getRect()).thenReturn(new Rectangle(point, contextDimension));
+        when(uiContext.getSearchContext()).thenReturn(searchContext);
         when(searchActions.findElements(locator)).thenReturn(List.of(element));
 
         touchSteps.swipeToElement(swipeDireciton, locator, Duration.ZERO);
         verifyNoInteractions(genericWebDriverManager);
     }
 
+    @Test
+    void shouldSwipeToElementIfItAlreadyExistsButOutOfArea()
+    {
+        WebElement element = mock(WebElement.class);
+        when(element.getRect()).thenReturn(new Rectangle(1, 1, 10, 10));
+        WebElement searchContext = mock(WebElement.class);
+        Point point = new Point(50, 0);
+        Dimension contextDimension = new Dimension(1920, 1080);
+        when(searchContext.getRect()).thenReturn(new Rectangle(point, contextDimension));
+        when(uiContext.getSearchContext()).thenReturn(searchContext);
+        mockAssertElementsNumber(List.of(element), true);
+        when(searchActions.findElements(locator)).thenReturn(List.of(element));
+        SearchParameters parameters = initSwipeMocks();
+        doAnswer(a ->
+        {
+            BooleanSupplier condition = a.getArgument(3, BooleanSupplier.class);
+            return null;
```

Comment:
what's the goal of this code?

Reply:
changed to
`        doNothing().when(touchActions).swipeUntil(eq(SwipeDirection.RIGHT), eq(Duration.ZERO), any(Rectangle.class),
                any(BooleanSupplier.class));`